vtm/src/org/oscim/renderer/bucket/ExtrusionBucket.java

```diff
@@ -184,50 +185,41 @@ public void addMesh(GeometryBuffer element) {
                 int vtx2 = index[k++] * 3;
                 int vtx3 = index[k++] * 3;
 
-                float vx1 = points[vtx1 + 0];
-                float vy1 = points[vtx1 + 1];
-                float vz1 = points[vtx1 + 2];
+                float[] v1 = new float[3];
```

Comment:
Can we have such arrays created once in class and filled each time needed, to avoid such object creations with each call?

Reply:
Do you really mean as private var or just outside the loop?
Can put it outside the loop, seems to be a minor difference: https://stackoverflow.com/questions/407255/difference-between-declaring-variables-before-or-in-loop

Reply:
Private final variable inside the class. May seems micro-optimization, but when handling many map elements we could check any such gain we can think.

Reply:
e.g. Viewport [temporary vars](https://github.com/mapsforge/vtm/blob/master/vtm/src/org/oscim/map/Viewport.java#L72-L75)